coriolis/tests/scheduler/rpc/test_client.py

```diff
@@ -0,0 +1,221 @@
+            mock.sentinel.ctxt, raise_if_none=False)
+        self.assertIsNone(result)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_any_worker_service_no_services(self, mock_get_workers):
+        mock_get_workers.return_value = []
+        self.assertRaises(
+            exception.NoWorkerServiceError,
+            self.client.get_any_worker_service, mock.sentinel.ctxt)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_any_worker_service_random_choice(self, mock_get_workers):
+        service_mock1 = {'id': 'test_id1'}
+        service_mock2 = {'id': 'test_id2'}
+        mock_get_workers.return_value = [service_mock1, service_mock2]
+
+        result = self.client.get_any_worker_service(
+            mock.sentinel.ctxt, random_choice=True)
+
+        self.assertIsInstance(result, dict)
+
+    @mock.patch.object(client.SchedulerClient, '_call')
+    def test_get_worker_service_for_specs(self, mock_call):
+        ctxt = mock.sentinel.ctxt
+        provider_requirements = mock.sentinel.provider_requirements
+        region_sets = mock.sentinel.region_sets
+        enabled = mock.sentinel.enabled
+        raise_on_no_matches = mock.sentinel.raise_on_no_matches
+
+        self.client.get_worker_service_for_specs(
+            ctxt, provider_requirements=provider_requirements,
+            region_sets=region_sets, enabled=enabled,
+            raise_on_no_matches=raise_on_no_matches)
+
+        mock_call.assert_called_once_with(
+            ctxt, 'get_workers_for_specs', region_sets=region_sets,
+            enabled=enabled, provider_requirements=provider_requirements)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_worker_service_for_specs_no_services_no_raise(
+            self, mock_get_workers):
+        mock_get_workers.return_value = []
+        result = self.client.get_worker_service_for_specs(
+            mock.sentinel.ctxt, raise_on_no_matches=False)
+        self.assertIsNone(result)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_worker_service_for_specs_no_services(self, mock_get_workers):
+        mock_get_workers.return_value = []
+        self.assertRaises(
+            exception.NoSuitableWorkerServiceError,
+            self.client.get_worker_service_for_specs, mock.sentinel.ctxt)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_worker_service_for_specs_random_choice(
+            self, mock_get_workers):
+        service_mock1 = {'id': 'test_id1'}
+        service_mock2 = {'id': 'test_id2'}
+        mock_get_workers.return_value = [service_mock1, service_mock2]
+
+        result = self.client.get_worker_service_for_specs(
+            mock.sentinel.ctxt, random_choice=True)
+
+        self.assertIsInstance(result, dict)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
```

Comment:
Also why mocking this? You should be mocking `get_worker_service_for_specs` directly.